File: TurfPro-main/lib/models/booking.dart
class Booking {
  final String userId;
  final String slotId;
  final String selectedSport;
  final DateTime selectedDate;
  final String selectedTimeSlot;
  final bool isFullCourt;
  final double subTotal;
  final double discount;
  final double payableAmount;
  final double advancePayment;

  Booking({
    required this.userId,
    required this.slotId,
    required this.selectedSport,
    required this.selectedDate,
    required this.selectedTimeSlot,
    required this.isFullCourt,
    required this.subTotal,
    required this.discount,
    required this.payableAmount,
    required this.advancePayment,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'slotId': slotId,
      'selectedSport': selectedSport,
      'selectedDate': selectedDate.toIso8601String(),
      'selectedTimeSlot': selectedTimeSlot,
      'isFullCourt': isFullCourt,
      'subTotal': subTotal,
      'discount': discount,
      'payableAmount': payableAmount,
      'advancePayment': advancePayment,
    };
  }
}

File: TurfPro-main/backend/models/booking.js
const mongoose = require('mongoose');

const bookingSchema = new mongoose.Schema({
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    slotId: { type: mongoose.Schema.Types.ObjectId, ref: 'Slot', required: true },
    selectedSport: { type: String, required: true },
    selectedDate: { type: Date, required: true },
    selectedTimeSlot: { type: String, required: true },
    isFullCourt: { type: Boolean, required: true },
    subTotal: { type: Number, required: true },
    discount: { type: Number, required: true },
    payableAmount: { type: Number, required: true },
    advancePayment: { type: Number, required: true },
}, { timestamps: true });

module.exports = mongoose.model('Booking', bookingSchema);
